spec/spec_helper_acceptance_local.rb

```diff
@@ -1,101 +1,64 @@
-  raise "stdout: #{first_apply[:stdout]}\nstderr: #{first_apply[:stderr]}" unless [0, 2].include?(first_apply[:exit_code])
-
-  second_apply = run_shell('sudo puppet agent -t --detailed-exitcodes', expect_failures: true)
-  raise "stdout: #{second_apply[:stdout]}\nstderr: #{second_apply[:stderr]}" unless second_apply[:exit_code] == 0
-end
-
-def apply_site_pp(manifest)
-  sitepp_content(manifest)
-
-  apply_result = run_shell('sudo puppet agent -t --detailed-exitcodes', expect_failures: true)
-  raise "stdout: #{apply_result[:stdout]}\nstderr: #{apply_result[:stderr]}" unless [0, 2].include?(apply_result[:exit_code])
-  block_given? ? yield(apply_result) : apply_result
-end
-
-def mockserver
-  @mockserver ||= Mockserver.new("#{ENV['TARGET_HOST']}:1080")
-  @mockserver
-end
-
-def set_default_api_mock
-  mockserver.set(default_endpoint, default_api_response, default_query_params)
-end
-
-def default_api_response
-  file_content = File.read('spec/support/files/valid_api_response.json')
-  file_content.gsub('dev84270.service-now.com', "#{ENV['TARGET_HOST']}:1080")
-end
-
-def default_endpoint
-  '/api/now/table/cmdb_ci'
-end
-
-def default_query_params
-  {
-    fqdn: ENV['TARGET_HOST'],
-    sysparm_display_value: 'true',
-  }
-end
-
-def servicenow_yaml_hash
-  yaml = run_shell('cat /etc/puppetlabs/puppet/servicenow.yaml')
-  YAML.safe_load(yaml[:stdout], symbolize_names: true)
-end
-
-def sitepp_content(manifest)
+# TODO: This will cause some problems if we run the tests
+# in parallel. For example, what happens if two targets
+# try to modify site.pp at the same time?
+def set_sitepp_content(manifest)
   content = <<-HERE
   node default {
     #{manifest}
   }
   HERE
 
-  run_shell("echo '#{content}' > /etc/puppetlabs/code/environments/production/manifests/site.pp")
+  on_master do |master|
+    master.run_shell("echo '#{content}' > /etc/puppetlabs/code/environments/production/manifests/site.pp")
+  end
 end
 
-def capture_trusted_notice(report)
-  # If you apply the Manifests::TRUSTED_EXTERNAL_VARIABLE manifest it will emit a notice
-  # that will contain the contents of the $trusted variable. This function will capture
-  json_regex = %r{defined 'message' as '(?<trusted_json>.+)'}
+def trigger_puppet_run(target, acceptable_exit_codes: [0, 2])
+  result = target.run_shell('puppet agent -t --detailed-exitcodes', expect_failures: true)
+  unless acceptable_exit_codes.include?(result[:exit_code])
+    raise "Puppet run failed\nstdout: #{result[:stdout]}\nstderr: #{result[:stderr]}"
+  end
+  result
+end
 
-  matches = report[:stdout].match(json_regex)
-  raise 'trusted external json content not found' if matches.nil?
-  parsed_data = JSON.parse(matches[:trusted_json], symbolize_names: true)
-  parsed_data.dig :external, :servicenow
+TRUSTED_JSON_SEPARATOR = '<TRUSTED_JSON>'.freeze
+def parse_trusted_json(puppet_output)
+  trusted_json = puppet_output.split(TRUSTED_JSON_SEPARATOR)[1]
```

Comment:
Forgot to explain why I changed this to use the separator. Reason is b/c it felt more accurate than https://github.com/puppetlabs/puppetlabs-servicenow_cmdb_integration/blob/master/spec/spec_helper_acceptance_local.rb#L95, which could potentially fail if (for some reason) Puppet decides to change the `defined <...>` part of the output.